path_metadata.go

```diff
@@ -288,6 +321,110 @@ func (b *versionedKVBackend) pathMetadataWrite() framework.OperationFunc {
 	}
 }
 
+// metadataPatchPreprocessor returns a framework.PatchPreprocessorFunc meant to
+// be provided to framework.HandlePatchOperation. The returned
+// framework.PatchPreprocessorFunc handles filtering out Vault-managed fields,
+// filtering out nulls, and ensuring appropriate handling of data types not
+// supported directly by FieldType.
+func metadataPatchPreprocessor() framework.PatchPreprocessorFunc {
```

Comment:
Why do we filter out nils here? Further down in `PathMetadataPatch`, the parsing logic should filter out nils already. Not a problem regardless as I think having the extra lair of defensive coding is a good thing, but I was just curious if I was missing something here.

Reply:
That's a good point. We have discussed this further offline and come to the conclusion that nils provided for top-level fields will be treated as unsetting the field. The [godoc](https://github.com/hashicorp/vault/blob/e54b5572ddc6d20a4f827662559c4586261e12b3/sdk/framework/backend.go#L281-L289) for `framework.HandlePatchOperation` will account for this functionality.